hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -510,6 +510,25 @@ private Object invokeOperation(Data key, MapOperation operation) {
         }
     }
 
+    protected InternalCompletableFuture<Data> putIfAbsentAsyncInternal(Object key, Data value,
+                                                                       long ttl, TimeUnit ttlUnit,
+                                                                       long maxIdle, TimeUnit maxIdleUnit) {
+        Data keyData = toDataWithStrategy(key);
```

Comment:
I am not sure how this internal API will be used. If it is used with a mixed 4.2-5.0 cluster (during rolling upgrade), then a serialization exception will be thrown by the 4.2 members. Probably it makes sense to gate the functionality on condition cluster version is at least 5.0 here, like this:

```
        if (isClusterVersionLessThan(Versions.V5_0)) {
            throw new UnsupportedOperationException("...");
        }
```

Reply:
@viliam-durina as rolling upgrade expert, what's your take on this?

Reply:
SQL doesn't support rolling upgrade from 4.2 to 5.0. But that doesn't mean that somebody will not attempt it. If the serialization exception thrown by 4.2 members is otherwise harmless, i'm fine with skipping this check.

Reply:
On second thought there is no new operation involved here, just new internal API that uses existing operations. So no serialization exception will happen.